### PR TITLE
Amy finish drag gameObject to be target gameObject sib

### PR DESCRIPTION
**Pull request by AmyOrz**

### CODE REVIEW

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+
+    switch (event |> ReactEventRe.Mouse.pageY) {
+    | pageY when pageY > domOffsetHeight + domOffsetTop - gapHeight =>
+      DragAfterTarget
+    | pageY when pageY < domOffsetTop + gapHeight => DragBeforeTarget
+    | pageY => DragIntoTarget
+    };
+  };
+
   let handleDragEnter =
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor: remove drag enter and event and reducer

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 2]
```diff
@@ -48,12 +73,43 @@ module Method = {
...
   };
 
-  let handleDrop =
+  let handleDragOver =
       (id, (isWidgetFunc, checkNodeRelationFunc, isAssetWDBFileFunc), event) => {
     let e = ReactEventType.convertReactMouseEventToJsEvent(event);
+    EventHelper.preventDefault(e);
+
+    let (isTrigger, _) =
+      DragEventBaseUtils.checkDragEnter(
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor: add _isValidForDragOver

let _isValidForDragOver = DragEventBaseUtils.isValidForDragEnter;

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 3]
```diff
@@ -48,12 +73,43 @@ module Method = {
...
+        id,
+        isWidgetFunc,
+        checkNodeRelationFunc,
+      );
+
+    let isSceneGameObject =
+      SceneEngineService.getSceneGameObject
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor to "id === SceneEngineService.getSceneGameObject |> StateLogicService.getEngineStateToGetData"

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 4]
```diff
@@ -32,7 +52,12 @@ module Method = {
         checkNodeRelationFunc,
       );
 
-    isTrigger || isAssetWDBFileFunc() ? DragEnter : Nothing;
+    isTrigger || isAssetWDBFileFunc() ?
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor(rename)(all: in sceneTree, asset, ...): rename DragEventBaseUtils.re->checkDragEnter,checkDragDrop to isValidForDragEnter/Drop

Comment (yyc-git) [reply src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor(rename): rename "isTrigger" to "isValid"

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 5]
```diff
@@ -102,34 +158,46 @@ module Method = {
       className=(
         "draggable-container"
         ++ (
-          isSelected ? isActive ? " select-active" : " select-not-active" : ""
+          isSelected ?
+            isActive ? " select-active " : " select-not-active " : " "
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor: extract buildMultipleClassName function

File: src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re [hunk 6]
```diff
@@ -127,26 +120,41 @@ module Method = {
...
   };
 
   let buildNotDragableUl = TreeNodeUtils.buildNotDragableUl;
 
+  let _buildMultipleClassName = (isSelected, isActive, dragGapClass) =>
+    "draggable-container"
+    ++ (
+      isSelected ? isActive ? " select-active " : " select-not-active " : " "
+    )
+    ++ dragGapClass;
```
Comment (yyc-git) [inline src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: TODO refactor: extract buildMultipleClassName function

Comment (AmyOrz) [reply src/core/atom_component/dragTree/component/treeNode/SceneTreeNode.re]: I already extract it.what you say?

File: src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re [hunk 7]
```diff
@@ -7,51 +7,132 @@ module CustomEventHandler = {
...
+    |> HierarchyGameObjectEngineService.setParentKeepOrder(
+         targetGameObjectUid,
+         draggedGameObjectUid,
+       );
+  };
 
-      | DragIntoTarget =>
-        SceneTreeEditorService.setIsShowChildren(targetUid, true)
-        |> StateLogicService.getAndSetEditorState;
+  let _handleDragToBeSceneGameObjectHierarchy =
```
Comment (yyc-git) [inline src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re]: TODO change to _handleDragToBeSceneGameObjectChild

File: src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re [hunk 8]
```diff
@@ -7,51 +7,132 @@ module CustomEventHandler = {
...
+         );
+    };
 
-      | DragAfterTarget =>
-        engineState
-        |> GameObjectUtils.setParentKeepOrder(
-             GameObjectUtils.getParentGameObject(targetUid, engineState)
-             |> OptionService.unsafeGet,
-             draggedUid,
+  let _handleDragToBeTargetGameObjectHierarchy =
```
Comment (yyc-git) [inline src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re]: TODO change to xxxSib

File: src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re [hunk 9]
```diff
@@ -7,51 +7,132 @@ module CustomEventHandler = {
...
+
+  let handleSelfLogic =
+      (
+        (store, dispatchFunc),
+        (),
+        (targetGameObjectUid, draggedGameObjectUid, dragPosition),
+      ) => {
+    let engineState = StateEngineService.unsafeGetState();
+
+    let isSceneGameObject =
```
Comment (yyc-git) [inline src/core/composable_component/mainEditor/composable_component/left_components/composable_component/sceneTree/eventHandler/SceneTreeDragGameObjectEventHandler.re]: TODO refactor(duplicate): extract isSceneGameObject func

**Pull request merged**